app/assets/src/components/ui/controls/SampleTypeSearchBox.jsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { groupBy } from "lodash/fp";
+
+import PropTypes from "~/components/utils/propTypes";
+
+import LiveSearchPopBox from "./LiveSearchPopBox";
+
+class SampleTypeSearchBox extends React.Component {
+  handleSearchTriggered = query => {
+    const matchSampleTypes = sampleType => {
+      // Match chars in any position. Good for acronyms.
+      const regex = new RegExp(query.split("").join(".*"), "gi");
+      if (regex.test(sampleType.name)) {
+        return true;
+      }
+      return false;
+    };
+    const matchedSampleTypes = this.props.sampleTypes.filter(matchSampleTypes);
+
+    const isSuggested = sampleType => {
+      // Insects only get their own types
+      if (this.props.isInsect) {
+        return sampleType.insect_only ? "suggested" : "donotshow";
+      }
+      // Humans get any type
+      if (this.props.isHuman) {
+        return "suggested";
+      }
+      // Non-human animals get suggested a subset
+      return sampleType.human_only ? "all" : "suggested";
+    };
+    const suggestedSampleTypes = groupBy(isSuggested, matchedSampleTypes);
+
+    return this.buildResults(suggestedSampleTypes, query);
+  };
+
+  buildResults(suggestedSampleTypes, query) {
+    const formatResult = result => {
+      return {
+        title: result.name,
+        name: result.name,
+        description: result.group,
+      };
+    };
+    const results = {};
+    if (suggestedSampleTypes.suggested) {
+      results.suggested = {
+        name: "SUGGESTED",
+        results: suggestedSampleTypes.suggested.map(formatResult),
+      };
+    }
+    if (suggestedSampleTypes.all) {
+      results.all = {
+        name: "ALL",
+        results: suggestedSampleTypes.all.map(formatResult),
+      };
+    }
+    // Always lowercase plain text input to distinguish it from Title Case
+    // offical sample types, which are enforced in SampleType.rb.
+    // TODO (gdingle): enforce lowercasing in CSV upload as well.
```

Comment:
not sure how best to do the same for CSV and CLI uploads in a simple way, and not sure if it's worth the trouble... @MarkAZhang ?